server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java

```diff
@@ -53,6 +56,13 @@ public TeamCityBuildListener(
     @Override
     public void buildStarted(@NotNull SRunningBuild build) {
         LOG.debug(String.format("Build started method triggered for %s, id %d", getBuildName(build), build.getBuildId()));
+
+        if (!nodesService.getCurrentNode().isMainNode())
+        {
+            LOG.debug(String.format("This is not the main node - skipping OTEL tracing for %s (id %d) and leaving that responsibility for the main node to deal with", getBuildName(build), build.getBuildId()));
+            return;
```

Comment:
Is there no way to just not load the build listener at all while retaining the UI extension?

Reply:
Nice idea - good thinking. 
I just went and investigated, and I dont think there is an easy way.

The DI container (spring) has this defined in xml, so while there's probably a fancy pants way of changing it all up so it only registers if it needs to, that feels like super advanced spring config to me, and that's a whole world of pain that I know very little about.